src/Controls/tests/DeviceTests/Elements/Window/WindowTests.Windows.cs

```diff
@@ -3,17 +3,64 @@
 using System.Threading.Tasks;
 using Microsoft.Maui.Controls;
 using Microsoft.Maui.DeviceTests.Stubs;
+using Microsoft.Maui.Graphics;
 using Microsoft.Maui.Graphics.Win2D;
 using Microsoft.Maui.Handlers;
 using Microsoft.Maui.Hosting;
 using Microsoft.Maui.Platform;
+using Microsoft.UI.Xaml.Automation.Peers;
+using Microsoft.UI.Xaml.Automation.Provider;
 using Xunit;
 using WPanel = Microsoft.UI.Xaml.Controls.Panel;
 
 namespace Microsoft.Maui.DeviceTests
 {
 	public partial class WindowTests : ControlsHandlerTestBase
 	{
+		[Fact(DisplayName = "Swapping MainPage no Crash")]
+		public async Task SwappingMainPageNoCrash()
+		{
+			SetupBuilder();
+
+			var button = new Button();
+
+			var mainPage = new ContentPage
+			{
+				BackgroundColor = Colors.Red,
+				Content = button
+			};
+
+			var secondaryPage = new ContentPage
+			{ 
+				BackgroundColor = Colors.Green 
+			};
+
+			var clicked = false;
+
+			button.Clicked += delegate
+			{
+				clicked = true;
+			};
+
+			await PerformClick(button);
```

Comment:
Why does this need a button click? Can we just wait for the layout and then perform the navigation?